src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.hpp

```diff
@@ -5,11 +5,41 @@
 #pragma once
 
 #include "jit_emitter.hpp"
+#include "emitters/utils.hpp"
 
 namespace ov {
 namespace intel_cpu {
 namespace aarch64 {
 
+template <dnnl::impl::cpu::aarch64::cpu_isa_t, typename>
+struct cpu_isa_vector_traits {};
+
+template <>
+struct cpu_isa_vector_traits<dnnl::impl::cpu::aarch64::cpu_isa_t::asimd, float16> {
+    typedef Xbyak_aarch64::VReg8H TReg;
+    typedef Xbyak_aarch64::VReg16B BReg;
+};
+
+template <>
+struct cpu_isa_vector_traits<dnnl::impl::cpu::aarch64::cpu_isa_t::asimd, float> {
+    typedef Xbyak_aarch64::VReg4S TReg;
+    typedef Xbyak_aarch64::VReg16B BReg;
+};
+
+// instead to have template class we have only one template method with dynamic template specialization by precision
+#define EMIT_IMPL() \
+void emit_impl(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs) const override { \
```

Comment:
Please move definition to cpp file.

Reply:
I can not. The marco was created for declaration & used in declaration.